js/plates-view/plate-mesh-fragment.glsl

```diff
@@ -84,9 +91,21 @@ void main() {
 	#include <logdepthbuf_fragment>
 	#include <map_fragment>
 	#include <color_fragment>
-    // --- CUSTOM:
-    diffuseColor *= vColor;
-    // ---
+
+  // --- CUSTOM:
+  if (vHidden == 1.0) {
+    // Hide pixel.
+    diffuseColor = vec4(0.0, 0.0, 0.0, 0.0);
+  } else if (vColor.a > 0.5) {
+    // Use specific color provided in color attribute. For example plate boundary color.
+    // Note that rendering only for alpha > X (e.g. 0.75 or 0.5) makes this color line thinner.
+    diffuseColor *= vec4(vColor.r, vColor.g, vColor.b, 1.0);
+  } else {
+    // Use the default colormap if vColor is transparent.
+    diffuseColor *= texture2D(colormap, vec2(0.5, vColormapValue));
```

Comment:
I am assuming that `vColor` is the vertex color and the `colorMap` sampler indexes into the UV map.

Just learning. :D

Reply:
> I am assuming that vColor is the vertex color

Yes, vColor is interpolated vertex color attribute. Note that we're in a fragment shader. This shader is run per each pixel on the screen. So, when you transition from vertex shader (run per each vertex only) to fragment (run per each pixel), vertex attributes are interpolated in a smart way.

>  the colorMap sampler indexes into the UV map.

Actually not. UV coords are not used here. colormap is a texture that has dimensions 1x<number_of_shades>. When you read its value using texture2D, you need to provide coords within range [0,1]. X is not important, so that's why I just use 0.5. And Y is the shade that we want to use. This is set using `vColormapValue`. It's an interpolated `colormapValue` vertex attribute that we set in JavaScript code. For a topo colorscale, it'd be normalized elevation ([0, 1] range so it works with texture).